server/pulp/server/webservices/views/tasks.py

```diff
@@ -67,6 +67,32 @@ def get(self, request):
         serialized_task_statuses = [task_serializer(task) for task in raw_tasks]
         return generate_json_response_with_pulp_encoder(serialized_task_statuses)
 
+    @auth_required(authorization.DELETE)
+    def delete(self, request):
+        """
+        Delete the tasks with status as finished, error, timed-out
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Status of delete operation or pulp Exception
+        :rtype:  String or pulp.Exception
+        """
+        task_state = request.GET.getlist('state')
+        if not task_state:
+            raise pulp_exceptions.PulpCodedForbiddenException()
+        complete_states = ("skipped", "error", "finished", "timed out")
+        for state in task_state:
+            if state not in complete_states:
+                raise pulp_exceptions.PulpCodedForbiddenException()
```

Comment:
Validation should happen before any deletion can happen. In this implementation, if three states are provided, and only the last is invalid, deletion will occur for the first two before the exception is raised.